internal/manifests/collector/servicemonitor_test.go

```diff
@@ -35,16 +37,34 @@ func TestDesiredServiceMonitors(t *testing.T) {
 	assert.Equal(t, fmt.Sprintf("%s-collector", params.OtelCol.Name), actual.Name)
 	assert.Equal(t, params.OtelCol.Namespace, actual.Namespace)
 	assert.Equal(t, "monitoring", actual.Spec.Endpoints[0].Port)
+	expectedSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "opentelemetry-collector",
+		"app.kubernetes.io/instance":   fmt.Sprintf("%s.%s", params.OtelCol.Namespace, params.OtelCol.Name),
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+		"app.kubernetes.io/part-of":    "opentelemetry",
+		"app.kubernetes.io/name":       naming.MonitoringService(params.OtelCol.Name),
+	}
+	assert.Equal(t, expectedSelectorLabels, actual.Spec.Selector.MatchLabels)
+}
 
-	params, err = newParams("", "testdata/prometheus-exporter.yaml")
+func TestDesiredServiceMonitorsWithPrometheus(t *testing.T) {
+	params, err := newParams("", "testdata/prometheus-exporter.yaml")
 	assert.NoError(t, err)
 	params.OtelCol.Spec.Observability.Metrics.EnableMetrics = true
-	actual, err = ServiceMonitor(params)
+	actual, err := ServiceMonitor(params)
 	assert.NoError(t, err)
 	assert.NotNil(t, actual)
 	assert.Equal(t, fmt.Sprintf("%s-collector", params.OtelCol.Name), actual.Name)
 	assert.Equal(t, params.OtelCol.Namespace, actual.Namespace)
 	assert.Equal(t, "monitoring", actual.Spec.Endpoints[0].Port)
 	assert.Equal(t, "prometheus-dev", actual.Spec.Endpoints[1].Port)
 	assert.Equal(t, "prometheus-prod", actual.Spec.Endpoints[2].Port)
+	expectedSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "opentelemetry-collector",
+		"app.kubernetes.io/instance":   fmt.Sprintf("%s.%s", params.OtelCol.Namespace, params.OtelCol.Name),
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+		"app.kubernetes.io/part-of":    "opentelemetry",
+		"app.kubernetes.io/name":       naming.MonitoringService(params.OtelCol.Name),
```

Comment:
@yuriolisa i realized that we actually need to update the monitoring service for the collector to have its own special label in the same way we do for headless and instead match on that.